### PR TITLE
delete cell values on drop column

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

**Comment by andrew-wm-arthur** [comment]
The primary objective here is to prevent cells that correspond to a tag/column that is not referenced in any schema. The scenario is: add column to working set, add values to column, drop column from working set. 

This implementation is more heavy-handed in that it always deletes values when we drop a column. If we want to be more careful will storage, we can only perform this pruning when we're deleting a column from the working set. 
@zachmu @bheni @reltuk

File: go/libraries/doltcore/schema/alterschema/dropcolumn.go [hunk 1]
```diff
@@ -71,15 +76,60 @@ func DropColumn(ctx context.Context, tbl *doltdb.Table, colName string) (*doltdb
...
+	mi, err := rowData.BufferedIterator(ctx)
+
+	if err != nil {
+		return types.EmptyMap, err
+	}
+
+	for {
+		k, v, err := mi.Next(ctx)
+
+		if k == nil || v == nil {
```
Comment (zachmu) [inline go/libraries/doltcore/schema/alterschema/dropcolumn.go]: Doesn't this iterator return err == io.EOF when it has no more elements? More common in go to match this error than check a nil value returned by the iterator.

File: go/libraries/doltcore/schema/alterschema/dropcolumn_test.go [hunk 2]
```diff
@@ -27,6 +27,46 @@ import (
 	"github.com/liquidata-inc/dolt/go/store/types"
 )
 
+var TypedRowsSansAge []row.Row
```
Comment (zachmu) [inline go/libraries/doltcore/schema/alterschema/dropcolumn_test.go]: s/sans/without/g

Get out of here with this francophile nonsense USA! USA!

Comment (andrew-wm-arthur) [reply go/libraries/doltcore/schema/alterschema/dropcolumn_test.go]: https://lh5.googleusercontent.com/proxy/UAFpmthJ6_r8i644McoXNJ7_kq1hY-q4IdTiHoSp3P2WjvBtP6ZsmA2dhBy7QQO_Jl4YZMteGF1Bf_EpdmBZlGddsCpMAGyeiYJG4eEIxw

File: go/libraries/doltcore/schema/alterschema/dropcolumn.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply go/libraries/doltcore/schema/alterschema/dropcolumn.go]: it doesn't return io.EOF the types package just gives `nil`

**Pull request merged**